src/Angor/Avalonia/Angor.Contexts.Funding/Founder/Operations/CreateProjectInfo.cs

```diff
@@ -0,0 +1,153 @@
+using Blockcore.NBitcoin.DataEncoders;
+using CSharpFunctionalExtensions;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using Stage = Angor.Shared.Models.Stage;
+
+namespace Angor.Contexts.Funding.Founder.Operations;
+
+public static class CreateProjectInfo
+{
+    public record CreateProjectInfoRequest(
+        WalletId WalletId, 
+        CreateProjectDto Project,
+        ProjectSeedDto ProjectSeedDto) : IRequest<Result<CreateProjectInfoResponse>>; // Returns project info event ID
+
+    public record CreateProjectInfoResponse(
+        string EventId);
+
+    public class CreateProjectInfoHandler(
+            ISeedwordsProvider seedwordsProvider,
+            IDerivationOperations derivationOperations,
+            IRelayService relayService,
+            IAngorIndexerService angorIndexerService,
+            IGenericDocumentCollection<DerivedProjectKeys> derivedProjectKeysCollection,
+            ILogger<CreateProjectInfoHandler> logger
+            ) : IRequestHandler<CreateProjectInfoRequest, Result<CreateProjectInfoResponse>>
+    {
+        public async Task<Result<CreateProjectInfoResponse>> Handle(CreateProjectInfoRequest request, CancellationToken cancellationToken)
+        {
+            if (request.ProjectSeedDto == null)
+            {
+                logger.LogDebug("FounderKeys is null in CreateProjectRequest for WalletId {WalletId}.", request.WalletId);
+                return Result.Failure<CreateProjectInfoResponse>("FounderKeys cannot be null.");
+            }
+
+            var wallet = await seedwordsProvider.GetSensitiveData(request.WalletId.Value);
+
+            ProjectSeedDto? newProjectKeys = request.ProjectSeedDto;
+          
+            var nostrPrivateKey = await derivationOperations.DeriveProjectNostrPrivateKeyAsync(wallet.Value.ToWalletWords(), newProjectKeys.FounderKey);
+            var nostrKeyHex = Encoders.Hex.EncodeData(nostrPrivateKey.ToBytes());
+
+            var projectInfoResult = await CreateProjectInfoOnNostr(nostrKeyHex, request.Project, newProjectKeys);
+
+            if (projectInfoResult.IsFailure)
+            {
+                logger.LogDebug("Failed to create project info on Nostr for Project {ProjectName} (WalletId: {WalletId}): {Error}", request.Project.ProjectName, request.WalletId, projectInfoResult.Error);
+                return Result.Failure<CreateProjectInfoResponse>(projectInfoResult.Error);
+            }
+
+            return Result.Success(new CreateProjectInfoResponse(projectInfoResult.Value));
+        }
+
+        private async Task<Result<string>> CreateProjectInfoOnNostr(string nostrKeyHex, CreateProjectDto project, ProjectSeedDto founderKeys)
+        {
+            var tsc = new TaskCompletionSource<Result<string>>();
+
+            ProjectInfo projectInfo;
+
+            switch (project.ProjectType)
+            {
+                case ProjectType.Invest:
+                    {
+                        projectInfo = new ProjectInfo
```

Comment:
Since you create a project info each time it's hard to tell what the differences actually are, perhaps creating a project info before the switch with all the common values and only setting the ones specific to each type in the switch would make it easier to understand.

Reply:
good idea, fixed.